### DISCUSSION TITLE
Problema con el task 03 Lab 3.01

### DISCUSSION THREAD

**Discussion by jfhr990120** [rramosp/ai4eng.v1#316] [03 - GENERACION DE MODELOS]
Hola, tengo un mensaje de error extraño en la plataforma cuando intento entregar el task 03

Mi código es este:

`` 
def SimpleModel():
    class _SimpleModel:

        def __init__(self):
            pass

        def fit(self, X, y):
            theta1=np.linspace(0, 1,11)
            theta2=np.linspace(0, 1, 11)
            mejor=0 #vamos a inicializar el método
            for i in theta1:
                for j in theta2:
                    t=np.array([i,j])
                    y_hat=predict(X,t)
                    accuracy=np.mean(y==y_hat)
                    if accuracy>mejor:
                        para=t
                        mejor=accuracy
            y_hat=predict(X,para)
            accuracy=np.mean(y==y_hat)
            t=para
            self.t=para
            return self

        def predict(self, X):
            tama=X.shape[0]
            d=np.ones(tama,dtype=int)
            indice=self.t
            for i in range(tama):
                if X[i][0]<indice[0] and X[i][1]>indice[1]:
                    d[i]=0
                else:
                    d[i]=1
        
            return d.T

    return _SimpleModel()
``

Y pues la plataforma me permite correrlo y evaluar los datasets de prueba de sklearn.

Pero a la hora de entregar la tarea, me sale este mensaje de error

![image](https://github.com/rramosp/ai4eng.v1/assets/162035477/4f032fc8-50cb-446c-addb-a22adc7eee8e)

Y pues ahí quedé mal porque no entiendo el error. Pareciera un error con la plataforma de calificación

@Felipe-RA 

Disculpas de antemano por las molestias

**Comment by Felipe-RA** [comment]
Saludos @jfhr990120 

Como bien indica el error,  tu fit esta usando un predict que no esta definido.

Si te funciona en los notebooks, probablemente se trata de un predict que tengas en memoria desde tu entorno de ejecución.

El problema esta en tu codigo

**Reply by jfhr990120** [reply]
Hola Felipe, muchas gracias por la respuesta

Lo resolví pero de una forma algo picapiedra

Lo que hice fue que en la función fit de la clase copié el código de la función predict para que hiciera todo allí
Pero pues esa solución no me satisface, creo que debe de haber una forma de poder llamar la función predict desde la función fit todo dentro de la clase del modelo

Sí se puede llamar una función desde otra función en la clase de sklearn???

Yo intenté hacerlo así

``
y=self.predict(self,X)
``
``
y=SimpleModel.predict(self,X)
``
Pero no pude llamarla

**Reply by Felipe-RA** [reply]
@jfhr990120  El motivo de tu problema es que el ambiente de prueba no tiene acceso a todo tu notebook, sólo a lo que dejes en la definición de la función, clase o variable que se envía.

Es por esto que se os suele pedir que todo import o función auxiliar se incluya como definición  dentro de la definición que es enviada

**Reply by jfhr990120** [reply]
Muchas gracias Felipe
